omise/omise.php

```diff
@@ -39,101 +54,68 @@ public function __construct()
-                ),
-                array(
-                    'type'     => 'text',
-                    'label'    => $this->l('Public key for live'),
-                    'name'     => 'publicKeyForLive',
-                    'required' => false,
-                    'desc'     => 'The "Live" mode public key can be found in Omise Dashboard.'
-                ),
-                array(
-                    'type'     => 'text',
-                    'label'    => $this->l('Secret key for live'),
-                    'name'     => 'secretKeyForLive',
-                    'required' => false,
-                    'desc'     => 'The "Live" mode secret key can be found in Omise Dashboard.'
-                ),
-                array(
-                    'label'    => '<b>Advance Settings</b>'
-                ),
-                array(
-                    'type'     => 'text',
-                    'label'    => $this->l('Title'),
-                    'name'     => 'title',
-                    'required' => false,
-                    'desc'     => 'This controls the title which the user sees during checkout.'
-                )
-            ),
-            'submit' => array(
-                'title' => $this->l('Save'),
-                'class' => 'btn btn-default pull-right'
-            )
-        );
-
-        $helper = new HelperForm();
-        $helper->submit_action = 'submit' . $this->name;
-
-        return $helper->generateForm($fields_form);
+        if ($this->setting->isSubmit()) {
+            $this->setting->save();
+            $this->smarty->assign('confirmation', $this->displayConfirmation($this->l('Settings updated')));
+        }
+
+        $this->smarty->assign('live_public_key', $this->setting->getLivePublicKey());
+        $this->smarty->assign('live_secret_key', $this->setting->getLiveSecretKey());
+        $this->smarty->assign('module_status', $this->setting->isModuleEnabled());
+        $this->smarty->assign('sandbox_status', $this->setting->isSandboxEnabled());
+        $this->smarty->assign('submit_action', $this->setting->getSubmitAction());
+        $this->smarty->assign('test_public_key', $this->setting->getTestPublicKey());
+        $this->smarty->assign('test_secret_key', $this->setting->getTestSecretKey());
+        $this->smarty->assign('title', $this->setting->getTitle());
+
+        return $this->display(__FILE__, 'views/templates/admin/setting.tpl');
+    }
+
+    public function hookPayment()
+    {
+        if ($this->active == false || $this->setting->isModuleEnabled() == false) {
+            return;
+        }
+
+        $this->smarty->assign('list_of_expiration_year', $this->checkout_form->getListOfExpirationYear());
+        $this->smarty->assign('omise_title', $this->setting->getTitle());
+
+        return $this->display(__FILE__, 'payment.tpl');
+    }
+
+    public function install()
+    {
+        if (parent::install() == false || $this->registerHook('payment') == false) {
```

Comment:
@guzzilar Done. 176057929efa72402b03b9c5d8095fa61a32b5ab

Refer to the third example in the section, **Using hooks**, on [PrestaShop Managing Hooks documentation](http://doc.prestashop.com/display/PS16/Managing+Hooks).

The condition has been changed to be more rigorous, both

1. Module installation (`parent::install()`)
2. Register a module to display the checkout form (`registerHook('payment')`)

must be success.

This change can prevent the impact in the step of module installation and the rare case so that the module uninstallation can be omitted.